tests/swagger/test_resources.py

```diff
@@ -1,104 +1,44 @@
-                }
-                resp = self.validate_request_call(op, expect_errors=True)
-
-                if op_id not in self.allowed_failures:
-                    # Method must exist (404 and 405 means it might not exist)
-                    self.assertIn(resp.status_int, [200, 201, 400])
-
-    def test_resource_utilities(self):
-        resource = self.resources['Utilities']
-        for op_id, op in resource.operations.items():
-                if op_id not in self.allowed_failures:
-                    self.validate_request_call(op)
-
-    def test_resource_batch(self):
-        resource = self.resources['Batch']
-        for op in resource.operations.values():
-            self.setUp()
-            requests = [{'path': '/v1/buckets'}]
-            defaults = {'method': 'POST'}
-            self.request.json = lambda: dict(
-                requests=requests,
-                defaults=defaults
-            )
-            self.validate_request_call(op)
-
-    def test_resource_buckets(self):
-        resource = self.resources['Buckets']
+                if op_id not in cls.allowed_failures:
+                    setattr(cls, 'test_resource_%s' % op_id,
+                            lambda self: cls.validate_request_call(self, op))
 
-        for op in resource.operations.values():
-            self.setUp()
-            self.request.path = {
-                'bucket_id': 'b1',
-            }
-            self.request.json = lambda: MINIMALIST_BUCKET
-            self.validate_request_call(op)
 
-    def test_resource_groups(self):
-        resource = self.resources['Groups']
+class SwaggerResourcesTest(six.with_metaclass(MetaResourcesTest, SwaggerTest)):
 
-        for op in resource.operations.values():
-            self.setUp()
-            self.request.path = {
-                'bucket_id': 'b1',
-                'group_id': 'g1',
-            }
-            self.request.json = lambda: MINIMALIST_GROUP
-            self.validate_request_call(op)
-
-    def test_resource_collections(self):
-        resource = self.resources['Collections']
+    allowed_failures = ['version']
 
-        for op in resource.operations.values():
-            self.setUp()
-            self.request.path = {
-                'bucket_id': 'b1',
-                'collection_id': 'c1',
-            }
-            self.request.json = lambda: MINIMALIST_COLLECTION
-            self.validate_request_call(op)
+    def validate_request_call(self, op, **kargs):
 
-    def test_resource_records(self):
-        resource = self.resources['Records']
+        self.request.url = op.path_name
 
-        for op in resource.operations.values():
-            self.setUp()
-            self.request.path = {
-                'bucket_id': 'b1',
-                'collection_id': 'c1',
-                'record_id': 'r1',
-            }
-            self.request.json = lambda: MINIMALIST_RECORD
-            self.validate_request_call(op)
+        request = self.request.__dict__
```

Comment:
I will patch this and I think I'll make `self.request` always a dict. I couldn't understand why, but `marshal_param` expects the request to be a dict while `validate_request` expects it to be a `IncomingRequest` object with attributes matching the fields used on `marshal_param` request dict. My guess is that  `IncomingRequest` was implemented later and they didn't want to change the param handling part of the code. I was using `validate_request` on the last version of test resources, but with it we needed to specify where each parameter would go by hand (e.g. body, path, header) now we are using only `marshal_param` to validate and set the parameters, which actually looks for the param specification and see where each param should go.